### ISSUE TITLE
Not working on Ubuntu 24

### ISSUE THREAD

**Issue by RaceNJason** [hjelev/rpi-mqtt-monitor#129]
I have this installed on several RPis which work great. I also have it installed on an Ubuntu desktop which also worked great up until the latest major release from v23 to v24. It didn't dawn on me that it was the upgrade that caused the break until I checked when the last time it had worked and when I had done the upgrade (I did the upgrade on July 7th). I verified that I have the latest monitor installed and also tried a restart of the service which succeeds, but still nothing coming through MQTT. If you are aware of something I could try to get it working again I would be grateful to hear it. On the other hand if it won't ever work on the latest (and future) versions of Ubuntu let me know that as well so I can uninstall it off the Linux desktop and not worry about it anymore. Thanks

**Comment by RaceNJason** [comment]
I should add that I did try a reinstall, but it didn't fix it. Also I have it set to run as a service. I didn't try the other option. And besides specifying my MQTT settings and it being a service, the rest I left to default values.

**Comment by JoshuaSeidel** [comment]
i get this issue as well, anyone figure it out?

**Comment by hjelev** [comment]
Can you show me some error message?
I have it running on Ubuntu 24.04 and 24.10 and there are no problems.

**Comment by RaceNJason** [comment]
Where would I find that? Basically I just don't get any data being sent to Home Assistant anymore even though it appears to be running on the Linux machine. Are there log files I can find on the Linux machine that would help?

**Comment by hjelev** [comment]
try running it like this rpi-mqtt-monitor from cmd.
If there is a problem with the connection you should see an error.
What the version configuration of the mqtt broker changed with the ubuntu upgrade or the upgrade was just for the machine running the monitor?

**Comment by RaceNJason** [comment]
OK...so here is the problem (and solution). Running it from the command line revealed a problem with the password. When I checked the raspberry pis the password in the config.py file were correct. But in the Linux build it wasn't correct even though I had put in the correct information (I know because I just did another clean install because of the recent version update). Anyway, the problem has to do with a password that has a '&' character in it. Basically with the Linux it translated my password as follows
"******u&Ra"
"******u"passwordRa"

Which gave the error of an unterminated string. I manually changed the password to the correct &Ra", did a sudo systemctl start rpi-mqtt-monitor.service command and it is working again. Why this Linux build wants to alter the string when an & symbol is in the password I'm not sure (especially when it doesn't occur on the RPi OS).

So I suspect all the other people that posted they are having the same problem probably have the & symbol in the password and the solution is to manually modify the config.py file in the src directory so that the password is correct.
